src/main/java/org/folio/consortia/service/impl/SharingSettingServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+import org.folio.consortia.domain.dto.PublicationRequest;
+import org.folio.consortia.domain.dto.SharingSettingRequest;
+import org.folio.consortia.domain.dto.SharingSettingResponse;
+import org.folio.consortia.domain.dto.Tenant;
+import org.folio.consortia.domain.dto.TenantCollection;
+import org.folio.consortia.domain.entity.SharingSettingEntity;
+import org.folio.consortia.repository.SharingSettingRepository;
+import org.folio.consortia.service.ConsortiumService;
+import org.folio.consortia.service.PublicationService;
+import org.folio.consortia.service.SharingSettingService;
+import org.folio.consortia.service.TenantService;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.scope.FolioExecutionContextSetter;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class SharingSettingServiceImpl implements SharingSettingService {
+
+  private final SharingSettingRepository sharingSettingRepository;
+  private final TenantService tenantService;
+  private final ConsortiumService consortiumService;
+  private final FolioExecutionContextHelper contextHelper;
+  private final PublicationService publicationService;
+  private final FolioExecutionContext folioExecutionContext;
+
+  @Override
+  @Transactional
+  public SharingSettingResponse start(UUID consortiumId, SharingSettingRequest sharingSettingRequest) {
+    UUID settingId = sharingSettingRequest.getSettingId();
+    log.debug("start:: Trying to share setting with consortiumId: {}, sharing settingId: {}", consortiumId, settingId);
+    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
+    Set<String> settingTenants = sharingSettingRepository.findTenantsBySettingId(settingId);
+    TenantCollection allTenants = tenantService.getAll(consortiumId);
+
+    PublicationRequest publicationPostRequest = createPublicationRequestForSetting(sharingSettingRequest, HttpMethod.POST.toString());
+    PublicationRequest publicationPutRequest = createPublicationRequestForSetting(sharingSettingRequest, HttpMethod.PUT.toString());
+
+    // By traverse through all tenants in db,
+    // we add tenant to put method publication tenant list, if it exists in setting tenant associations
+    // otherwise, we add it to post method publication tenant list
+    for (Tenant tenant : allTenants.getTenants()) {
+      if (settingTenants.contains(tenant.getId())) {
+        publicationPutRequest.getTenants().add(tenant.getId());
+        log.info("start:: tenant={} added to publication update request for setting={}", tenant.getId(), settingId);
```

Comment:
you also need to update existing setting ids in database to have correct metadata when they updated